src/otx/core/data/transform_libs/torchvision.py

```diff
@@ -2380,19 +2403,29 @@ def _crop_data(
 
         img: np.ndarray = to_np_image(inputs.image)
         orig_shape = inputs.img_info.img_shape
-        margin_h = max(orig_shape[0] - crop_size[0], 0)
-        margin_w = max(orig_shape[1] - crop_size[1], 0)
-        offset_h, offset_w = self._rand_offset((margin_h, margin_w))
-        crop_y1, crop_y2 = offset_h, offset_h + crop_size[0]
-        crop_x1, crop_x2 = offset_w, offset_w + crop_size[1]
+        crop_bbox, offset_h, offset_w = self._generate_crop_bbox(orig_shape, crop_size)
```

Comment:
It seems that this line doesn't have to be exceuted for semantic segmentation task. How about to change code to skip this line if task is semantic segmentation?

Reply:
No, it was changed for semantic segmentation because `crop_bbox` is updated until getting desirable cropped region within 10 times.
https://github.com/openvinotoolkit/training_extensions/blob/b8307a6be314352ba2b90aa73acb09f09b131af5/src/otx/core/data/transform_libs/torchvision.py#L2406C9-L2418
Of course, eventually, I don't use `RandomCrop` for semantic segmentation due to performance issue.